### DISCUSSION TITLE
When server restarts, client auto-reconnection only works well for some clients

### DISCUSSION THREAD

**Discussion by qiulang** [socketio/socket.io#3850] [Q&A] [answered]
My socket.io client only uses websocket (using ` transports: ['websocket']`) . Through the log we collected we found that when the socket.io server restarts, some clients reconnect quickly, i.e. auto-reconnection works well for them. But some clients waited from 10 minutes to even 1 or 2 hours to reconnect. 

My clients are in difference cities so I can't just say they all had bad network connection (until I have solid proof for that)

So any suggestion how to debug this issue 

I use "socket.io-client": "~2.1.0"  and "socket.io": "^2.3.0",

**Comment by darrachequesne** [comment, marked as answer]
That sounds indeed weird. Are you sure they do not lose network connectivity (which is common for mobile users)?

**Reply by qiulang** [reply]
They are not mobile users as we develop desktop webapp. 

I just found you had changed socket.io-client 2.3 to use debug 3.1. That was the reason I used 2.1 (to support IE 11) so I plan to upgrade socket.io-client to 2.3 to see if that would help.

BTW, does 3.0 or 4.0 add any enhancement for auto-reconnection ? I read the release notes but I didn't see they mentioned that.

**Reply by darrachequesne** [reply]
I don't think there was any change to the auto-reconnection in v3 or v4 (the implementation is rather basic).

If you have additional details, or a way to reproduce, that would be great :pray:

**Reply by qiulang** [reply]
Hi I finally found the reason of my problem. The auto-reconnection works as expected, sorry for the false alarm.
The reason I had thought the it didn't work was because I run socket.io server in docker and there were several docker contains running. But the reason I had not figured out yet, k8s sometimes killed and restarted one of those dockers but not all.

So I saw some clients reconnected and some didn't. Those didn't was because they connected to the docker containers that did not restart!

Sorry again for the false alarm. :$
